src/com/sap/piper/analytics/Analytics.groovy

```diff
@@ -0,0 +1,106 @@
+package com.sap.piper.analytics
+
+import com.cloudbees.groovy.cps.NonCPS
+import org.jenkinsci.plugins.workflow.steps.MissingContextVariableException
+
+class Analytics implements Serializable{
+
+    private static Analytics instance
+
+    private List eventListeners
+
+    private Analytics(){
+        this.eventListeners = []
+    }
+
+    @NonCPS
+    static Analytics getInstance(){
+        if(!instance) {
+            createInstance()
+
+            Closure defaultListener = {payload ->
+                piperOsDefaultReporting(payload)
+            }
+            registerEventListener(defaultListener)
+        }
+        return instance
+    }
+
+    static void createInstance(){
+        instance = new Analytics()
+    }
+
+    static void registerEventListener(Closure listener){
```

Comment:
We could add another method `registerEventListener(Class listenerClass)` or similar to register by Class/ClassName.

Reply:
I would keep it simple for now and only provide one possibility (the one with the Closure).